Ran DamMet last night, success!!
However, only a fraction of CpGs actually were able to provide enough data to perform methylation estimation. Need to assess reasons why...
- coverage/depth? Either sites were not covered at all or not covered with enough depth

How can I find out stats on coverage and depth...

do 
`samtools stats in.bam`
`samtools coverage `
`samtools flagstat` for quick information

How to interpret samtools flagstats?
- there's # + #...what does this mean?
	- first number is reads passing filters, second is reads failing filters
- first line tells you total # of reads going in, then later there is # reads mapped (this is the important one)
- what does 'properly paired' mean?
	- Sounds like this is no. reads where they match up with each other and both map well to the same area (i.e., it's not that one read maps and the other doesn't)

Yes....coverage was way too low to estimate anything, average coverage was less than 1X (should have run gargammel with -c for coverage instead of no. of fragments, I guess this tells me I need to enrich but I think I knew that already...likely no chance of looking at methylation with shotgun data)