Today's goals: 
- [ ] Align USER-treated sequences to human reference genome using bwa with ancient settings
- [ ] Generate mapDamage plot from USER-treated aligned sequences
- [ ] Put both mapDamage plots into report

Aligning USER-treated sequences
`cd /2/scratch/natassja/Bio722/term_project/bioinformatic_user
`bwa aln GRCH38.p14.genome.fa -n 0.01 -o 2 -l 16500 -t 10 /2/scratch/natassja/Bio722/term_project/gargammel/quality_control/sim_seqs_s1_trimmed_USER.fq.gz > sim_seqs_s1_trim_USER_aln.sai

(& same for s2)
Then do: 
`bwa sampe GRCh38.p14.genome.fa sim_seqs_s1_trim_USER_aln.sai sim_seqs_s1_trim_USER_aln.sai sim_seqs_s1_trimmed_USER sim_seqs_s2_trimmed_USER > sim_seqs_trim_USER_aln.sam `

Convert SAM to BAM
`samtools view -b sim_seqs_trim_USER_aln.sam > sim_seqs_trim_USER_aln.bam`

Assessing damage patterns after USER treatment
`mapDamage -i sim_seqs_trim_USER_aln.bam -r /2/scratch/natassja/Bio722/gargammel/alignment/ref_db/GRCh38.p14.genome.fa`
