2024-02-22 
- gargammel successfully ran with default parameters on GRCh human reference genome
- better understanding of conda
- opened Obsidian notebook and figured out how to update to Github

2024-03-08
- Ran gargammel to generate simulated sequence data

2024-03-27
- Ran bwa mem to align simulated sequences to human reference genome, generate SAM file

2024-03-27
- Ran mapDamage2 to generate misincorporation plot

2024-04-01
- Installed DamMet and attempted to install epiPALEOMIX
- asked for help on installing older version of R for epiPALEOMIX

2024-04-05
- Installed older version of R in epipaleomix-env
- EpiPaleomix up and running!

2024-04-10
- Created and edited YAML makefile template for epiPaleomix
- Tried first run of epiPaleomix

2024-04-13
- Thinking about bioinformatic USER
- Tried fqtools & ran MapDamage2
- Troubleshooting odd output from mapDamage2

2024-04-20
- Ran successful dry run of epiPaleomix 
- Troubleshooting YAML file
- Troubleshooting DamMet

2024-04-23
- Re-ran gargammel with more accurate parameters
- Re-aligned using bwa aln instead of bwa mem
- Ran MapDamage2 again

2024-04-24
- Re ran bioinformatic USER treatment 
- Ran mapDamage on USER treated sequences
- Ran DamMet on USER and non USER treated sequences

2024-04-25
- Ran samtools (depth, coverage, flagstat) to get information on mapping quality 
