## Quality control with fastp
- before mapping with bwa, I need to first trim simulated adapters using fastp
- do I have paired-end or single-end "reads"? How can I find this out?
- I have paired end reads because I have two files output (s1 and s2)

### fastp usage

Ran 
`fastp -i /home/natassja/scratch/Bio722/term_project/gargammel/results/sim_data_s1.fq.gz -I /home/natassja/scratch/Bio722/term_project/gargammel/results/sim_data_s2.fq.gz -o sim_data_s1 -O sim_data_s2
`
Basic fastp run with default settings, no merging
All reads passed filters, all adapters trimmed

## Mapping with bwa
- bwa has more than one alignment algorithm
- inputs necessary: reference genome, fastq file with sequences
- bwa mem performs local alignments using maximal exact matches
- bwa aln just finds the coordinates of the input reads
- for anything, you first need to index the reference genome using `bwa index ref.fa`

Step 1: index reference genome with bwa index
Ran:
`ln -s /home/natassja/scratch/Bio722/term_project/gargammel/data/endo/GRCH38.genome.fa input`
`bwa index input`

