Today's goals: 
- [x] Align USER-treated sequences to human reference genome using bwa with ancient settings
- [x] Generate mapDamage plot from USER-treated aligned sequences
- [ ] Put both mapDamage plots into report

Aligning USER-treated sequences
`cd /2/scratch/natassja/Bio722/term_project/bioinformatic_user
`bwa aln GRCH38.p14.genome.fa -n 0.01 -o 2 -l 16500 -t 10 /2/scratch/natassja/Bio722/term_project/gargammel/quality_control/sim_seqs_s1_trimmed_USER.fq.gz > sim_seqs_s1_trim_USER_aln.sai

(& same for s2)
Then do: 
`bwa sampe GRCh38.p14.genome.fa sim_seqs_s1_trim_USER_aln.sai sim_seqs_s1_trim_USER_aln.sai sim_seqs_s1_trimmed_USER sim_seqs_s2_trimmed_USER > sim_seqs_trim_USER_aln.sam `

Convert SAM to BAM
`samtools view -b sim_seqs_trim_USER_aln.sam > sim_seqs_trim_USER_aln.bam`

Assessing damage patterns after USER treatment
`mapDamage -i sim_seqs_trim_USER_aln.bam -r /2/scratch/natassja/Bio722/gargammel/alignment/ref_db/GRCh38.p14.genome.fa`

We would expect that after USER treatment, we would see both the mean insert size decrease and the damage patterns lower significantly at the ends of molecules. 

Hmm... only the 5' end shows a difference in the damage patterns. Also need to trim the 3' with fqtools. 

New approach - use seqkit trimfq function

`seqkit trimfq -b [left trim] -e [right trim] in.fq > out.fq`

Ran bwa again...while this is running try to get DamMet to work

Trying...
`DamMet estDeam -b sim_aln_sorted.bam -r reference -c reference.fai -B /home/natassja/sequences/cpgs_all.bed`

Error: NOT BED COORDS ON CHROM

BWA finished
Sort BAM files

`samtools sort in.bam > out_sorted.bam`
`samtools index sorted.bam`

Use UCSC tools to get the bed coordinates of human chromosomes

`conda install -c bioconda -n ucsc-tools-env ucsc-fetchchromsizes`
`fetchChromSizes hg38 | awk 'OFS="\t"{print $1, "0", $2}' > hg38.bed`

`for chr in ``cat chromosomes.txt```; do DamMet estDeam -b sim_seqs_trim_USER_aln_sorted.bam -r GRCh38.p14.genome.fa -c $chr -W 30 -O USER_$chr; done

To look at specific regions I'm interested in: 

`for chr in ``cat chromosomes.txt```; do DamMet estDeam -b sim_seqs_s1s2_trim_aln_sorted.bam -r reference -c $chr -B /home/natassja/sequences/cpgs_all.bed -O sites_$chr; done

(& run for USER treated sequences)

Can I put all the deam and F files together?? Concatenate? Append?
How can I compare USER and non-USER treated sequences directly (DamMet says this is possible) and how can I present this graphically?

Yes, concatenated all .F files with the following:

`for chr in ```cat chromosomes.txt```; do cat sites_$chr.BED.F >> sites_ALL.BED.F; done
(& same for USER treated)

Move all loci that have data (i.e., not CpG NOT FOUND) to a new file with: 

`grep 'N_CpGs sites_ALL.BED.F > sites_ALL_found.BED.F`
(& same for USER-treated sequences)






