# Working on a bioinformatic USER treatment

- need to differentiate between partial and full USER (partial targets uracils in middle of fragment, full targets ones on the end)
- aDNA is often single-stranded on ends and the majority of deamination damage to unmethylated cytosines happens here
- When a uracil is excised from a single stranded region close to the end of the molecule 

Thinking about how to make this work...need a way to read the 5' and 3' misincorporation files, retrieve the position number for the line that has the frequency that drops below the threshold, then trim all the fq sequences that number of positions

Possibly simpler to do this with bash scripting...using awk? Then something like fqtools trim?

Trimmomatic? No...will only trim based on quality scores

Arguments to pass to script
- 5' misincorporation file
- 3' misincorporation file
- threshold (can I set a default?)

Can do this less elegantly using awk and fqtools..

To get the cutoff point from the misincorporation files: 

`awk -F' ' '$2<0.01{print$1}' 5pCtoT_freq.txt | head -1`
`awk -F' ' '$2<0.01{print$1}' 3pGtoA_freq.txt | head -1`

Then do: 

`conda activate fqtools-env`
`fqtools trim -o sim_data_s1_USER -s 7 sim_data_s1_trimmed.fastq.gz`

`fqtools trim -o sim_data_s2_USER -s 7 sim_data_s2_trimmed.fastq.gz`

Have generated USER-treated files for s1 and s2! Need to confirm this worked as expected by generating another mapDamage plot
- first need to align sequences to generate BAM file

This is what I ran before to align the simulated sequences: 

` bwa mem GRCh38.p14.genome.fa /2/scratch/natassja/Bio722/term_project/gargammel/quality_control/sim_data_s1_trimmed.fq /2/scratch/natassja/Bio722/term_project/gargammel/quality_control/sim_data_s2_trimmed.fq > sim_aln.sam

Run this again with the new files: 
WAIT

I need to USER treat the already trimmed files, not the raw data files
Try this one again...

Re-ran on sim_data_s1_trimmed_fq.gz and sim_data_s2_trimmed.fq.gz
Generated files: 
- sim_data_s1_trimmed_USER.fq.gz
- sim_data_s2_trimmed_USER.fq.gz

NOW run bwa mem again (indexed reference files located in /2/scratch/natassja/Bio722/term_project/gargammel/alignment/ref_db)

- Took about 4sec to generate file sim_aln_USER.sam

Run mapDamage using: 

`conda activate mapdamage-env`
`mapDamage -i sim_aln_USER.sam -r GRCh38.p14.genome.fa`

mapDamage gave warning - damage levels too low - is this a feature or a bug....
it's still running though so maybe this is what we were looking for

Plot looks a little odd: 

![](Pasted%20image%2020240420074139.png)

So the deamination damage frequencies are now very low (expected), but the frequencies of each base have changed also (also expected...?)

For reference here is the original plot from mapDamage2

![](Pasted%20image%2020240420074327.png)

I think something went wrong...while the frequencies of damage should be low beyond the 7bp trimmed section, it shouldn't be zero. Looks like all the complexity is lost - again this shouldn't happen because the aim is just to trim off the first 7 bases of each read

What did fqtools actually do?

`fqtools trim -h` returned: 

View FASTQ files.

view options:
  -h               Show this help message and exit.
  -k               Preserve secondary headers (if present).
  -o STEM          Output file stem (default "output%").
  -s LENGTH        Trim LENGTH bases from the read start.
  -l LENGTH        Trim the read to a maximum length of LENGTH.
  FILE             The fastq file(s) to view.

STEM:
    The file stem to use for output files (without file extension). Any
    instances of the single character specified using the -p global argument
    will be replaced with the pair number, or removed for single output
    files. If the -o option is not specified, single file output will
    be written to stdout and paired file output to the default stem (output%).

FILE:
    If no input file is specified, input will be read from stdin.

LENGTH:
    If LENGTH is not specified, no trimming is performed.


Should be correct? I used -s instead of -l to trim 7 bases from the read start instead of to the maximum length of 7. 

Ran 
`gunzip -c sim_data_s1_trimmed_USER.fastq.gz > sim_data_s1_trimmed_USER.fastq`

To look directly at fastq files. Yes, they have been trimmed correctly
Maybe the problem is with bwa and how the sequences are aligning. It's possible that some are too short to align properly at 28bp long (compared to 35 bp for the original sequences)

Also may be better to use bwa aln instead of bwa mem since these sequences are shorter

