State of the project...still attempting to troubleshoot both epiPaleomix and DamMet. Project due in three days. 
- drawbacks to using these programs - no help available on StackExchange, biostars, etc. Documentation for the programs likely assumes some kind of prior knowledge that I don't have. 
- possibility that I can figure this out in time, also possible that I cannot
- Have been working on this since the start of the month (installed both programs April 1)

## Revisiting gargammel after USER results

If you recall the results of mapDamage after USER treatment indicated there was something going wrong, possibly due to the static length chosen when running gargammel. What if we ran this again with better parameters?

Instead of generating fragments that are all the same length (in this case, 35), let's try and generate fragments that actually reflect a typical fragment length distribution (should probably have done this earlier)

### Getting a realistic FLD tab file
- use Kingston samples!
- I've performed preprocessing on a set of samples from a cemetery in Kingston Ontario and have FLDs for all these samples
- Choose one FLD file (will be some source of bias based on this particular sample but better than choosing just 35)
- Use the FLD file when running gargammel

### Getting methylation-specific deamination damage
- I realized recently that I didn't run gargammel correctly the first time
- I did specify deamination damage, but it didn't simulate differential deamination damage due to methylation (need the --methyl flag AND a matrix file specifying rate)
- Getting the matrix file specifying rates...precalculated rates of deamination exist and can be found in one of the files (path src/matrices)

### Also changing...
- number of reads...I know I was trying to simulate a regular shotgun run, but I think in order to do this analysis properly I need more reads (which we would address in the lab anyways by)

Revamped gargammel run:
`gargammel -o results/sim_seqs -f FLD.tab -n 1000000 -damage 0.024,0.36,0.0097,0.68 --methyl -matfilemeth /home/sam/gargammel/src/matrices/double- -matfilenonmeth /home/sam/gargammel/src/matrices/double- data 

Worked! 

Ran fastqc
` fastqc /2/scratch/natassja/Bio722/term_project/gargammel/results/sim_seqs_s1.fq.gz --outdir .`
(same for s2)

Ran fastp (v 0.23.2)
`fastp -i /2/scratch/natassja/Bio722/term_project/gargammel/results/sim_seqs_s1.fq.gz -I /2/scratch/natassja/Bio722/term_project/gargammel/results/sim_seqs_s2.fq.gz -o sim_seqs_s1_trimmed -O sim_seqs_s2_trimmed`

Quality control done - files for next steps are

/2/scratch/natassja/Bio722/term_project/gargammel/quality_control/sim_seqs_s1_trimmed.fq.gz
/2/scratch/natassja/Bio722/term_project/gargammel/quality_control/sim_seqs_s2_trimmed.fq.gz

## Aligning and assessing damage

Non-user treated samples first: (bwa version 0.7.17-r1188)
`bwa aln GRCH38.p14.genome.fa -n 0.01 -o 2 -l 16500 -t 10 /2/scratch/natassja/Bio722/term_project/gargammel/quality_control/sim_seqs_s1_trimmed.fq.gz > sim_seqs_s1_trim_aln.sai
(& run again for s2)

Flag explanation: 
- -n 0.01 is maximum edit distance
- -o 2 is max gaps
- -l 16500 is...

Output of bwa aln is in SAI format, run: 
`bwa sampe GRCh38.p14.genome.fa sim_seqs_s1_trim_aln.sai sim_seqs_s2_trim_aln.sai sim_seqs_s1_trimmed sim_seqs_s2_trimmed > sim_seqs_trim_aln.sam `

Convert SAM to BAM
`samtools view -b sim_seqs_s1s2_trim_aln.sam > sim_seqs_s1s2_trim_aln.bam`

Run MapDamage2
`mapDamage -i sim_seqs_s1s2_trim_aln.bam -r GRCh38.p14.genome.fa`

Ran awk on 5 prime and 3 prime conversion frequency files

`awk -F' ' '$2<0.02{print$1}' 5pCtoT_freq.txt | head -1 `
`awk -F' ' '$2<0.02{print$1}' 3pGtoA_freq.txt | head -1 `

Ran fqtools on both s1 and s2 files to create USER-treated files

`fqtools trim -o sim_seqs_s1_trimmed_USER -s 7 sim_seqs_s1_trimmed`
(& same for s2)





